MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -16762,8 +16775,19 @@ void ApplicationWindow::checkForProjectRecovery() {
   }
 
   // Recovery file present
-  if (m_projectRecovery.attemptRecovery()) {
-    // If it worked correctly reset project recovery and start saving
+  try {
+    m_projectRecovery.attemptRecovery();
+  } catch (std::exception &e) {
+    std::string err{
+        "Project Recovery failed to recover this checkpoint. Details: "};
+    err.append(e.what());
+    g_log.error(err);
+    QMessageBox::information(this, "Could Not Recover",
+                             "We could not fully recover your work.\nMantid "
+                             "will continue to run normally now.",
```

Comment:
Can we make clearer here that as we could not fully recover we are in fact doing no recovery and juts starting from a clean Mantid.